ddprofiler/app/src/test/java/ddprofiler/test/PreAnalyzerTest.java

```diff
@@ -32,6 +46,83 @@ public class PreAnalyzerTest {
     private String username = "root";
     private String password = "Qatar";
 
+    /**
+     * Setup PreAnalyzer before each test
+     * (specifically for new tests, not legacy ones).
+     */
+    @Before
+    public void setUp() throws CsvValidationException, SQLException, IOException {
+        // Avoid Exceptions by adding a sample column to the source.
+        when(source.getAttributes()).thenReturn(new ArrayList<>(Arrays.asList(new Attribute("sample"))));
+
+        // Initialize and set source of PreAnalyzer
+        preAnalyzer = new PreAnalyzer(profilerConfig);
+        preAnalyzer.assignSourceTask(source);
+    }
+
+    /**
+     * Test readRows method for spatial data, ensuring correct
+     * semantic types and granularity.
+     */
+    @Test
+    public void testReadRowsSpatialData() throws CsvValidationException, SQLException, IOException {
+        // Prepare sample spatial columns
+        Map<Attribute, List<String>> spatialData = getSpatialData();
+        when(source.readRows(1)).thenReturn(spatialData);
+        preAnalyzer.readRows(1);
+
+        // Ensuring correct semantic type & granularity
+        for (Entry<Attribute, List<String>> dataEntry : spatialData.entrySet()) {
+            Attribute attribute = dataEntry.getKey();
+            System.out.println(attribute.getColumnName());
```

Comment:
Since we already have assertions, we don't need additional print statements. If you feel it is still necessary for debugging purposes, I think the statements should be printed in a clearer format.

Reply:
Thanks for pointing it out! It was meant for testing purposes previously, and it has been removed in the latest commit.